__tests__/app.test.js

```diff
@@ -84,3 +80,49 @@ test("404 : Responds with message -'Not Found' when article id is valid but does
       expect(body.msg).toBe("Not Found");
     });
 });
+
+describe("GET /api/articles/:article_id/comments", () => {
+  test("200: Responds with an array of article comments for the specified article id", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments).toBeSortedBy("created_at", { descending: true });
+        comments.forEach((comment) => {
+          expect(comment.article_id).toBe(1);
+          expect(comment).toHaveProperty("comment_id", expect.any(Number));
+          expect(comment).toHaveProperty("votes", expect.any(Number));
+          expect(comment).toHaveProperty("created_at", expect.any(String));
+          expect(comment).toHaveProperty("author", expect.any(String));
+          expect(comment).toHaveProperty("body", expect.any(String));
+        });
+      });
+  });
+  test("200: Responds with an empty array if article_id exists but there are no comments ", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments.msg).toBe("No comments found");
+      });
+  });
```

Comment:
Great testing! 👍  Could we also just add another expect to this 200 test, which shows the comments are an empty array please?

Reply:
HI, thank you for the feedback.  im not sure how to do this because the response which comes back is an object and i cant find the property would show the empty array. the only way i know is to add a comment poperty onto here: 
   return Promise.resolve({ status: 200, comments: [], msg: "No comments found" });?

Reply:
this seemed to solve it .... Am i okay to merge like this? thanks 

  exports.selectArticleComments = (article_id) => {
  const query =
    "SELECT * FROM comments WHERE article_id = $1 ORDER BY comments.created_at DESC";

  return db.query(query, [article_id]).then(({ rows }) => {
    if (!rows.length) {
      return Promise.resolve({ comments: `${rows}`,status: 200, msg: "No comments found" });
    }

    return rows;
  });
};test("200: Responds with an empty array if article_id exists but there are no comments ", () => {
    return request(app)
      .get("/api/articles/2/comments")
      .expect(200)
        .then(({body}) => {
        const { comments } = body;
        expect(comments.comments).toHaveLength(0)
        expect(comments.msg).toBe("No comments found");
      });
  });